SofaKernel/modules/SofaDeformable/SofaDeformable_test/PolynomialRestShapeSpringsForceField_test.cpp

```diff
@@ -0,0 +1,70 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <SofaSimulationGraph/testing/BaseSimulationTest.h>
+using sofa::helper::testing::BaseSimulationTest;
+
+#include <SofaSimulationGraph/SimpleApi.h>
+using sofa::simulation::Node;
+
+#include <sofa/defaulttype/Vec.h>
+using sofa::defaulttype::Vec3;
+
+class PolynomialRestShapeSpringsForceField_test : public BaseSimulationTest
+{
+public:
+    /// Run one stepsof simulation then check results
+    bool testSpringForce()
+    {
+        double dt = 0.01;
+        sofa::simpleapi::importPlugin("SofaComponentAll") ;
```

Comment:
The tests should include corner cases to insure the component is behaving well  (so working in a meaninfull manner and not crashing) when users sets bad inputs in the data field.